aicsimageio/tests/readers/test_array_like_reader.py

```diff
@@ -1132,3 +1133,88 @@ def test_aicsimage_from_array(
             dict,
         ),
     )
+
+
+@pytest.fixture
+def arange_data() -> list[xr.DataArray]:
+    dims = list("STCZYX")
+    shape = (3, 4, 5, 6, 7, 8)
+    data = np.arange(np.prod(shape), dtype="uint16").reshape(shape)
+    coords = {
+        "S": [f"Image:{i}" for i in range(shape[0])],
+        "C": ["bf", "dapi", "gfp", "yfp", "dsred"],
+    }
+    return [arr.drop_vars("S") for arr in xr.DataArray(data, dims=dims, coords=coords)]
+
+
+def test_get_stack_defaults(arange_data: list[xr.DataArray]) -> None:
+    alr = ArrayLikeReader(arange_data)
+    data = xr.concat(arange_data, dim="I")
+
+    # numpy
+    stack = alr.get_stack()
+    assert np.allclose(stack, data)
+
+    # dask
+    dask_stack = alr.get_dask_stack()
+    assert np.allclose(dask_stack, data).compute()
+
+    # xarray dask
+    xarray_dask_stack = alr.get_xarray_dask_stack()
+    xr.testing.assert_allclose(xarray_dask_stack, data)
+
+    # xarray numpy
+    xarray_stack = alr.get_xarray_stack()
+    xr.testing.assert_allclose(xarray_stack, data)
+
+
+def test_get_stack_different_shape(arange_data: list[xr.DataArray]) -> None:
+    arange_data[1] = arange_data[1][:3, :4, :5, :6]
+    alr = ArrayLikeReader(arange_data)
+    data = xr.concat(arange_data[::2], dim="I")
+
+    with pytest.raises(exceptions.UnexpectedShapeError):
+        alr.get_stack()
+
+    stack = alr.get_stack(drop_non_matching_scenes=True)
+    np.testing.assert_allclose(data.data, stack)
+
+    stack = alr.get_stack(select_scenes=(0, 2))
+    np.testing.assert_allclose(data.data, stack)
+
+    stack = alr.get_xarray_stack(drop_non_matching_scenes=True)
+    xr.testing.assert_allclose(data, stack)
+
+
+def test_get_stack_different_dtype(arange_data: list[xr.DataArray]) -> None:
+    arange_data[1] = arange_data[1].astype("int64")
+    alr = ArrayLikeReader(arange_data)
+    data = xr.concat(arange_data[::2], dim="I")
+
+    with pytest.raises(TypeError):
+        alr.get_stack()
+
+    stack = alr.get_stack(drop_non_matching_scenes=True)
+    np.testing.assert_allclose(stack, data.data)
+
+    stack = alr.get_stack(select_scenes=(0, 2))
+    np.testing.assert_allclose(stack, data.data)
+
+    stack = alr.get_xarray_stack(drop_non_matching_scenes=True)
+    xr.testing.assert_allclose(stack, data)
+
+
+def test_get_stack_scene_coords(arange_data: list[xr.DataArray]) -> None:
+    alr = ArrayLikeReader(arange_data)
+    data = xr.concat(arange_data, dim="I")
+    data = data.assign_coords({"I": [f"Image:{i}" for i in range(3)]})
+
+    stack = alr.get_xarray_stack(scene_coord_values="names")
+    xr.testing.assert_allclose(stack, data)
+
+    # test different scene char
+    stack = alr.get_xarray_stack(scene_coord_values="names", scene_character="S")
```

Comment:
the use of "S" here really should be abolished because aicsimageio reserves it for "samples" as in "rgb".

Reply:
This is a test for a _different_ scene character than default "I" i think its fine? but if you wanted like "give me a random char" maybe `U` - i cant think of any known dim that uses the U character

Reply:
Old habits die hard haha. But I will eliminate all use of `S` throughout here. I'd personally like to keep `scene_character` as a kwarg but whatever you all want.

Reply:
I would also like to keep a kwarg. I am not sure what dim characters are out there and I am worried that `I` may be out there somewhere already. Better to just have it as a parameter.

Reply:
My point here is what if the image file has RGB samples AND multiple scenes, and you say scene_character="S"?